tofu/spectro/_rockingcurve.py

```diff
@@ -0,0 +1,345 @@
+
+    # Atomic number of Si and O atoms
+    Zsi = 14.
+    Zo = 8.
+
+    # Position of the three Si atoms in the unit cell (page 242 Wyckoff)
+    u = 0.4705
+    xsi = np.r_[-u, u, 0.]
+    ysi = np.r_[-u, 0., u]
+    zsi = np.r_[1./3., 0., 2./3.]
+
+    # Position of the six O atoms in the unit cell (page 242 Wyckoff)
+    x = 0.4152
+    y = 0.2678
+    z = 0.1184
+    xo = np.r_[x, y-x, -y, x-y, y, -x]
+    yo = np.r_[y, -x, x-y, -y, x, y-x]
+    zo = np.r_[z, z+1./3., z+2./3., -z, 2./3.-z, 1./3.-z]
+
+    # Bragg angle and atomic plane distance d for a given wavelength lamb and
+    # Miller index (h,k,l)
+    d_num = np.sqrt(3.)*a0*c0
+    d_den = np.sqrt(4.*(ih**2 + ik**2 + ih*ik)*c0**2 + 3.*il**2*a0**2)
+    if d_den == 0.:
+        msg = (
+            "Something went wrong in the calculation of d, equal to 0!\n"
+            "Please verify the values for the following Miller indices:\n"
+            + "\t - h: first Miller index ({})\n".format(ih)
+            + "\t - k: second Miller index ({})\n".format(ik)
+            + "\t - l: third Miller index ({})\n".format(il)
+        )
+        raise Exception(msg)
+    else:
+        d_atom = d_num/d_den
+    if d_atom < lamb/2.:
+        msg = (
+            "According to Bragg law, Bragg scattering need d > lamb/2!\n"
+            "Please check your wavelength arg.\n"
+        )
+        raise Exception(msg)
+    else:
+        sol = 1./(2.*d_atom)
+        sin_theta = lamb/(2.*d_atom)
+        theta = np.arcsin(sin_theta)
+        theta_deg = theta*180./np.pi
+    lc = [theta_deg < 10., theta_deg > 89.]
+    if any(lc):
+        msg = (
+            "The computed value of theta is behind the arbitrary limits.\n"
+            "Limit condition: 10° < theta < 89° and\n"
+            "theta = ({})°\n".format(theta_deg)
+        )
+        raise Exception(msg)
+
+    # Atomic scattering factors ["asf"] for Si(2+) and O(1-) as a function of
+    # sol = sin(theta)/lambda ["sol_values"], taking into account molecular
+    # bounds
+    # From Henry & Lonsdale, "International tables for Crystallography" (1969)
+    # Vol.III p202 or Vol.IV page 73 for O(1-), Vol.III p202 ? for Si(2+)
+    sol_si = np.r_[
+        0., 0.1, 0.2, 0.25, 0.3, 0.35, 0.4, 0.5, 0.6,
+        0.7, 0.8, 0.9, 1., 1.1, 1.2, 1.3, 1.4, 1.5,
+    ]
+    asf_si = np.r_[
+        12., 11., 9.5, 8.8, 8.3, 7.7, 7.27, 6.25, 5.3,
+        4.45, 3.75, 3.15, 2.7, 2.35, 2.07, 1.87, 1.71, 1.6,
+    ]
+    sol_o = np.r_[
+        0., 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1., 1.1,
+    ]
+    asf_o = np.r_[
+        9., 7.836, 5.756, 4.068, 2.968, 2.313, 1.934, 1.710, 1.566, 1.462,
+        1.373, 1.294,
+    ]
+
+    fig = plt.figure(figsize=(8, 6))
+    gs = gridspec.GridSpec(1, 1)
+    ax = fig.add_subplot(gs[0, 0])
+    ax.set_xlabel(r'sin($\theta$)/$\lambda$')
+    ax.set_ylabel("atomic scattering factor")
+    ax.plot(sol_si, asf_si, label="Si")
+    ax.plot(sol_o, asf_o, label="O")
+    ax.legend()
```

Comment:
Add a `plot_atomic_scattering_factor=None` as an input argument of the method,
Set it to True by default
If True => call a sub-routine `_plot_atomic_scattering_factor(sol_si, asf_si, sol_o, asf_o)` dedicated to making this plot

General idea: plots are nice, but they should be:
* optional (=> user can activate a keyword argument to request the plot, or deactivate it)
* treated in dedicated sub-routines (to keep the core of the function cleaner and simpler)